pkg/apis/wksprovider/controller/wksctl/machine_controller.go

```diff
@@ -203,7 +202,8 @@ func (a *MachineController) connectTo(ctx context.Context, c *existinginfrav1.Ex
 }
 
 func (a *MachineController) sshKey(ctx context.Context) ([]byte, error) {
-	secret, err := a.clientSet.CoreV1().Secrets(a.controllerNamespace).Get(ctx, controllerSecret, metav1.GetOptions{})
+	var secret v1.Secret
```

Comment:
shall we be consistent in passing secrets as pointers everywhere and make this `secret := &v1.Secret{}` instead?
Also, I'm not sure what happens when you pass the client a nil pointer tbh. Maybe it works.

Reply:
I've updated these to use stack-allocated variables where possible as per our discussion. This eliminates the need to deal with potential `nil`s. Looking at the implementation of client it doesn't seem like passing `nil` would work anyways.